src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/AbstractRazorDelegatingEndpoint.cs

```diff
@@ -52,6 +52,16 @@ internal abstract class AbstractRazorDelegatingEndpoint<TRequest, TResponse> : I
     /// </summary>
     protected virtual bool PreferCSharpOverHtmlIfPossible { get; } = false;
 
+    /// <summary>
+    /// When <see langword="true" />, we'll try to send request to Html even when it is in a Razor context, for example
+    /// for component attributes that are a Razor context context, but we want to treat them as Html for auto-inserting quotes
+    /// after typing equals for attribute values.
+    /// </summary>
+    internal virtual Task<bool> PreferHtmlOverRazorIfPossibleAsync(TRequest request, RazorRequestContext requestContext, DocumentPositionInfo positionInfo, CancellationToken cancellationToken)
```

Comment:
There is an existing mechanism we should be able to use to get this info, rather than have this method. The `DocumentPositionInfoStrategy` property allows inheritors to specify a service to use to resolve position info, which includes language kind.

Reply:
Thanks, makes sense. Please see next commit. If that's what you had in mind, I will add tests in the next commit.

Reply:
Yep, looks perfect. I would have hoped the original tests would still work though 🤷‍♂️

I didn't actually look at any code in detail though

Reply:
Thanks! Sadly, original tests were unit tests for the new helper method I added. Since that went away, the unit tests will be for the new service instead. The code will essentially be the same, just changing what's being tested.